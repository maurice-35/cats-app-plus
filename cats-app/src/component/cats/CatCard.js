import React, { useState, useRef } from "react";
import { Card, Col, Popover, Button, Overlay } from "react-bootstrap";

const CatCard = ({
  image,
  origin,
  name,
  description,
  energy_level,
  dog_friendly,
  affection_level,
  temperament,
  life_span,
}) => {
  const [show, setShow] = useState(false);
  const [target, setTarget] = useState(null);

  const ref = useRef(null);

  const handleClick = (event) => {
    setShow(!show);
    setTarget(event.target);
  };

  return (
    <div className="container2">
      <Card.Header>{name}</Card.Header>
      <Card.Img variant="top" src={image ? image.url : ""} alt={name} />
      <Card.Body>
        <Card.Title id="show">{origin}</Card.Title>
      </Card.Body>
      <Col>
        <Card.Body>
          <Card.Title id="show">{name}</Card.Title>
          <div>
            <div ref={ref}>
              <Card.Text>
                Description
                <Button
                  variant="outline-success"
                  id="button"
                  onClick={handleClick}
                >
                  Click me!
                </Button>
                <Overlay
                  show={show}
                  target={target}
                  placement="bottom"
                  container={ref}
                  containerPadding={20}
                >
                  <Popover id="popover-contained">
                    <Popover.Header as="h3">Hello!</Popover.Header>
                    <Popover.Body>
                      <strong>{description}</strong>
                    </Popover.Body>
                  </Popover>
                </Overlay>
              </Card.Text>
            </div>
          </div>
        </Card.Body>
        <Card.Footer>
          <small className="text">Energy-level: {energy_level}</small>
          <br />
          <small className="text">Dog friendly: {dog_friendly}</small>
          <br />
          <small className="text">Affection-level: {affection_level}</small>
          <br />
          <small className="text">Temperament: {temperament}</small>
          <br />
          <small className="text">Life-span: {life_span}</small>
          <br />
          <small className="text">Origin: {origin}</small>
          <br />
        </Card.Footer>
      </Col>
    </div>
  );
};

export default CatCard;
