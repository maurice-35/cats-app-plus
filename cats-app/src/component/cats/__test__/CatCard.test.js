import { render, screen, fireEvent } from '@testing-library/react'
import { act } from "react-dom/test-utils";
import CatCard from '../CatCard'


const mockedSetNavigation = jest.fn()

describe("CatCard", () => {
	test("Button should display description", () => {
		render(<CatCard />);
		const displayDescription = screen.getByText(/Click me!/i);
		expect(displayDescription).toMatchSnapshot();
	});


	it("should display cats page", () => {
		render(
			<CatCard
				show={[]}
				setShow={mockedSetNavigation}
			/>
		)
		const displayDescription = screen.getByText(/Click me!/i)
		fireEvent.click(displayDescription)
		fireEvent.change(displayDescription, { target: { value: 'description' } })
		const buttonElement = screen.getByRole('button', { name: /Click me!/i })
		fireEvent.click(buttonElement)
		// eslint-disable-next-line testing-library/no-unnecessary-act
		act(async () => {
			fireEvent.click(displayDescription)
		})
		expect(displayDescription).toMatchSnapshot()
	})
})
