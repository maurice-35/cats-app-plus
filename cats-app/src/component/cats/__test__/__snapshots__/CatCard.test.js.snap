// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatCard Button should display description 1`] = `
<button
  class="btn btn-outline-success"
  id="button"
  type="button"
>
  Click me!
</button>
`;

exports[`CatCard should display cats page 1`] = `
<button
  aria-describedby=""
  class="btn btn-outline-success"
  id="button"
  type="button"
  value="description"
>
  Click me!
</button>
`;
